ui/helpers/utils/notification.util.ts

```diff
@@ -317,20 +318,16 @@ export function getNetworkDetailsByChainId(chainId?: keyof typeof CHAIN_IDS): {
   const nativeCurrencyAddress = '0x0000000000000000000000000000000000000000';
 
   const blockExplorerConfig =
-    ETHERSCAN_SUPPORTED_NETWORKS[
-      chainId as keyof typeof ETHERSCAN_SUPPORTED_NETWORKS
+    SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS[
+      chainId as keyof typeof SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS
```

Comment:
Nit: There is a type mismatch here that we're overwriting. The input `chainId` is optional, and the type implies that it can have a a far wider variety of values than is present in `SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS`.

The type of this `blockExplorerConfig` object should be `typeof SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS[keyof typeof SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS] | undefined`, but the `| undefined` part is currently missing because of this cast. This is OK in this case because the return type allows it, but we're losing that protection against breaking this in a future change because we're misleading `tsc`, meaning it won't be able to help us prevent a regression here (e.g. it won't guard against making this output property mandatory).

I'd recommend that we fix this with a type guard function. I've experimented with it locally, and this seems to work:

```
/**
 * Type guard to ensure a key is present in an object.
 *
 * @param object - The object to check.
 * @param key - The key to check for.
 * @returns True if the key is present, false otherwise.
 */
function isKey(object: RuntimeObject, key: PropertyKey): key is keyof object {
  return hasProperty(object, key);
}

...

const blockExplorerConfig =
    chainId && isKey(SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS, chainId)
      ? SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS[chainId]
      : undefined;
```

Though note that I also had to add an `as const` on the definition of `SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS` to get the type inference to work properly (we're using this as a constant here anyway).

See here for more on the risks of using type assertions, and why using type inference and type guards improves type safety: https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#avoid-type-assertions-by-using-type-guards-to-improve-type-inference

Reply:
Valid. I'm mostly fine with the internal implementation as you've declared the return type for this property is optional (`| undefined`).

Feel free to use a `isKey` if this exists, or use the `'key' in` syntax. This is good enough since this is a const object.

```ts
const blockExplorerConfig = chainId in SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS
  ? SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS[chainId]
  : undefined;
```

Reply:
That's what I tried first, but TypeScript didn't seem to infer the type as we'd expect. if that works, even better.

Reply:
Thanks for the suggestion!
I've added the `isKey` type guard function to ensure we're correctly checking for the presence of keys in the `SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS` object. I also updated the definition of `SUPPORTED_NOTIFICATION_BLOCK_EXPLORERS` with as const to improve type inference